From: Carsten Schoenert <c.schoenert@t-online.de>
Date: Fri, 23 Dec 2016 18:13:38 +0100
Subject: OpenSSL: Make SSLv2 conditional

OpenSSL1.1 doesn't support SSLv2 anymore, make the code conditional so
we can build against both OpenSSL versions.
Patch based on work done for Dovecot.
---
 common/ECChannel.cpp                       | 4 ++++
 provider/server/ECSoapServerConnection.cpp | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/common/ECChannel.cpp b/common/ECChannel.cpp
index cab099f4..17f2f0fb 100644
--- a/common/ECChannel.cpp
+++ b/common/ECChannel.cpp
@@ -95,12 +95,16 @@ HRESULT ECChannel::HrSetCtx(ECConfig *lpConfig, ECLogger *lpLogger) {
 			ssl_neg = true;
 		}
 
+#ifdef SSL_TXT_SSLV2
 		if (strcasecmp(ssl_name, SSL_TXT_SSLV2) == 0)
 			ssl_proto = 0x01;
 		else if (strcasecmp(ssl_name, SSL_TXT_SSLV3) == 0)
+#else
+		if (strcasecmp(ssl_name, SSL_TXT_SSLV3) == 0)
 			ssl_proto = 0x02;
 		else if (strcasecmp(ssl_name, SSL_TXT_TLSV1) == 0)
 			ssl_proto = 0x04;
+#endif
 #ifdef SSL_TXT_TLSV1_1
 		else if (strcasecmp(ssl_name, SSL_TXT_TLSV1_1) == 0)
 			ssl_proto = 0x08;
diff --git a/provider/server/ECSoapServerConnection.cpp b/provider/server/ECSoapServerConnection.cpp
index 4e5a49d6..c69f56f1 100644
--- a/provider/server/ECSoapServerConnection.cpp
+++ b/provider/server/ECSoapServerConnection.cpp
@@ -249,12 +249,16 @@ ECRESULT ECSoapServerConnection::ListenSSL(const char* lpServerName, int nServer
 			ssl_neg = true;
 		}
 
+#ifdef SSL_TXT_SSLV2
 		if (strcasecmp(ssl_name, SSL_TXT_SSLV2) == 0)
 			ssl_proto = 0x01;
 		else if (strcasecmp(ssl_name, SSL_TXT_SSLV3) == 0)
+#else
+		if (strcasecmp(ssl_name, SSL_TXT_SSLV3) == 0)
 			ssl_proto = 0x02;
 		else if (strcasecmp(ssl_name, SSL_TXT_TLSV1) == 0)
 			ssl_proto = 0x04;
+#endif
 #ifdef SSL_TXT_TLSV1_1
 		else if (strcasecmp(ssl_name, SSL_TXT_TLSV1_1) == 0)
 			ssl_proto = 0x08;
